Classes/Hooks/CmdMapLocalizationHook.php

```diff
@@ -0,0 +1,207 @@
+                    $originalElement[$field],
+                    $targetLanguage['language_isocode'],
+                    $sourceLanguage['language_isocode']
+                );
+                if (!empty($translatedContent) && isset($translatedContent['translations'])) {
+                    foreach ($translatedContent['translations'] as $translation) {
+                        if ($translation['text'] != '') {
+                            $value = $translation['text'] ?? $value;
+                            $deepLTranslated = true;
+                            break;
+                        }
+                    }
+                }
+                $translatedElement[$field] = $value ?? $originalElement[$field];
+            }
+            if ($deepLTranslated) {
+                // empty slug field to auto create
+                if ($detectedSlugField !== '') {
+                    $translatedElement[$detectedSlugField] = '';
+                }
+                $data[$table][$translatedElement['uid']] = $translatedElement;
+                $innerDataHandler = GeneralUtility::makeInstance(DataHandler::class);
+                $innerDataHandler->start($data, []);
+                $innerDataHandler->process_datamap();
+                GeneralUtility::makeInstance(PageRepository::class)
+                    ->markPageAsTranslatedWithDeepl($siteInformation['pageUid'], $targetLanguage);
+            }
+        }
+    }
+
+    private function cleanUpWithDefault(
+        string $table,
+        array $originalElement,
+        array $translatedElement,
+        DataHandler $dataHandler,
+        array $tableTCA,
+        array $targetLanguage
+    ): void {
+        // load some defaults
+        $pageId = $table === 'pages' ? $originalElement['uid'] : $originalElement['pid'];
+        $TSConfig = BackendUtility::getPagesTSconfig($pageId)['TCEMAIN.'] ?? [];
+        $tableEntries = $dataHandler->getTableEntries($table, $TSConfig);
+        if (!empty($TSConfig['translateToMessage']) && !($tableEntries['disablePrependAtCopy'] ?? false)) {
+            $translateToMsg = $this->getLanguageService()->sL($TSConfig['translateToMessage']);
+            $translateToMsg = @sprintf($translateToMsg, $targetLanguage['title']);
+        }
+
+        $detectedSlugField = '';
+        foreach ($translatedElement as $field => $value) {
+            if (
+                isset($tableTCA[$field])
+                && isset($tableTCA[$field]['config']['type'])
+                && $tableTCA[$field]['config']['type'] === 'slug'
+            ) {
+                $detectedSlugField = $field;
+                continue;
+            }
+            if (
+                !isset($tableTCA[$field])
+                || !isset($tableTCA[$field]['config']['type'])
+                || (
+                    $tableTCA[$field]['config']['type'] !== 'text'
+                    && $tableTCA[$field]['config']['type'] !== 'input'
```

Comment:
Would replace this with a `in_array()` instead of repeating the array access path here

```php
|| !in_array($tableTCA[$field]['config']['type'], ['text', 'input'])
```